pkg/resource/deploy/step.go

```diff
@@ -979,9 +1000,9 @@ func (s *RefreshStep) Persisted() bool { return s.cts != nil }
 
 func (s *RefreshStep) Op() display.StepOp                           { return OpRefresh }
 func (s *RefreshStep) Deployment() *Deployment                      { return s.deployment }
-func (s *RefreshStep) Type() tokens.Type                            { return s.old.Type }
-func (s *RefreshStep) Provider() string                             { return s.old.Provider }
-func (s *RefreshStep) URN() resource.URN                            { return s.old.URN }
```

Comment:
Instead of adding `typ`, `prov` and `urn` to the `RefreshStep`, couldn't we just return `s.new.Type`, etc. here? If I'm following the code correctly that should amount to the same thing.

Reply:
No because new might be changed to nil later and then these are still accessed after that :( we could have origionalNew and then actualNew.